### PR TITLE
TST: Use pytest or decorator instead of checking ImportError

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

### CODE REVIEW

File: pandas/tests/frame/test_reductions.py [hunk 1]
```diff
@@ -285,16 +285,11 @@ def test_stat_op_api(self, float_frame, float_string_frame):
...
-            from scipy.stats import (  # noqa:F401
-                kurtosis,
-                skew,
-            )
-
-            assert_stat_op_api("skew", float_frame, float_string_frame)
-            assert_stat_op_api("kurt", float_frame, float_string_frame)
-        except ImportError:
-            pass
+    @pytest.mark.filterwarnings("ignore:Dropping of nuisance:FutureWarning")
```
Comment (MarcoGorelli) [inline pandas/tests/frame/test_reductions.py:288]: why is this necessary?

**Review by MarcoGorelli — COMMENTED** [review]
Nice! Just got a question

File: pandas/tests/frame/test_reductions.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/frame/test_reductions.py:288]: I'm getting
```
pandas/tests/frame/test_reductions.py::TestDataFrameAnalytics::test_stat_op_api_skew_kurt
  /pandas/tests/frame/test_reductions.py:160: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
    getattr(float_string_frame, opname)(axis=0)
```

without it. Copying it from the original larger test, guessing it's based on the fixture data and `assert_stat_op_api`

Comment (MarcoGorelli) [reply pandas/tests/frame/test_reductions.py:288]: Ah, I hadn't noticed it was part of the original test, thanks for explaining

**Review by MarcoGorelli — APPROVED** [review]
looks good to me, seems like a nice cleanup
